--- a/modules/ui/workspaces/autoload/workspaces.el
+++ b/modules/ui/workspaces/autoload/workspaces.el
@@ -545,7 +545,11 @@ This be hooked to `projectile-after-switch-project-hook'."
       (unwind-protect
           (if (and (not (null +workspaces-on-switch-project-behavior))
                    (or (eq +workspaces-on-switch-project-behavior t)
-                       (equal (safe-persp-name (get-current-persp)) persp-nil-name)
+                       ;; HACK Perspective main
+                       (let ((current-persp-name (safe-persp-name (get-current-persp))))
+                         (or (equal current-persp-name persp-nil-name)
+                             (equal current-persp-name "main")))
+                       ;; (equal (safe-persp-name (get-current-persp)) persp-nil-name)
                        (+workspace-buffer-list)))
               (let* ((persp
                       (let ((project-name (doom-project-name +workspaces--project-dir)))
